### ISSUE TITLE
Relative Markdown links not working

### ISSUE THREAD

**Issue by karmeye** [arnau/obsidian-metatable#47]
Two issues regarding Markdown links (not wiki) when the link format is set to *Relative path to file*. (Not sure if they apply to other link formats as well.)

## 1. File Not Found

Obsidian automatically inserts `%20` for spaces in file names, which is needed for Markdown links not wrapped with `""`. In the YAML, however, the link is necessarily wrapped already and because `%20` is inserted the link will not work when clicked on, you get: Folder does not exist (i.e., the file doesn’t).

Would it be possible to prevent obsidian from doing this?

## 2. Clicking Jumps to Wrong File

This one occurs when there are several files with the same name on different levels in the folder hierarchy. For example, I have ReadMe.md files in every folder:

```
/ReadMe
/a/ReadMe
/a/b/ReadMe
/a/b/c/File #Linking to /a/b/ReadMe
```

When clicking on the link in the YAML of File, /ReadMe at the root is opened. If I rename it to something that doesn’t match the link, then /a/ReadMe is opened instead. Only after I also rename this file, the expected file is opened.

So it seems like the plugin is ignoring the `../` in the link and just searches the folder path from the root down and returns the first match.

**Comment by arnau** [comment]
@karmeye thanks for reporting, I think this is a regression. I'll have a look.

**Comment by arnau** [comment]
On (2 Cliking jumps to the wrong file), I'm afraid the issue comes from Obsidian itself:

https://forum.obsidian.md/t/absolute-or-relative-path-default-location-for-new-notes-should-be-ignored-when-creating-files-by-following-a-link-the-link-itself-should-be-used/41743

**Comment by arnau** [comment]
@karmeye version 0.14.6 should fix most if not all the issues you reported here.

1. The `%20` issue should be fixed.
2. The jump to the wrong file is an Obsidian known issue. However, I made a change that attempts to expand to the full path which should mitigate the problem. Let me know if it misbehaves.
3. Fixed. However it mimics Obsidian's behaviour so if you want to guarantee it links to the file in the same folder use an explicit `./`.
